tests/Gedmo/Tree/MaterializedPathUuidORMTest.php

```diff
@@ -0,0 +1,172 @@
+        static::assertSame(1, $category->getLevel());
+        static::assertSame(2, $category2->getLevel());
+        static::assertSame(3, $category3->getLevel());
+        static::assertSame(1, $category4->getLevel());
+
+        static::assertSame($this->getTreeRootValueOfRootNode($category), $category->getTreeRootValue());
+        static::assertSame($this->getTreeRootValueOfRootNode($category2), $category2->getTreeRootValue());
+        static::assertSame($this->getTreeRootValueOfRootNode($category3), $category3->getTreeRootValue());
+        static::assertSame($this->getTreeRootValueOfRootNode($category4), $category4->getTreeRootValue());
+
+        // Update
+        $category2->setParent(null);
+
+        $this->em->persist($category2);
+        $this->em->flush();
+
+        $this->em->refresh($category);
+        $this->em->refresh($category2);
+        $this->em->refresh($category3);
+
+        static::assertSame($this->generatePath([$category->getId()]), $category->getPath());
+        static::assertSame($this->generatePath([$category2->getId()]), $category2->getPath());
+        static::assertSame($this->generatePath([$category2->getId(), $category3->getId()]), $category3->getPath());
+        static::assertSame(1, $category->getLevel());
+        static::assertSame(1, $category2->getLevel());
+        static::assertSame(2, $category3->getLevel());
+        static::assertSame(1, $category4->getLevel());
+
+        static::assertSame($this->getTreeRootValueOfRootNode($category), $category->getTreeRootValue());
+        static::assertSame($this->getTreeRootValueOfRootNode($category2), $category2->getTreeRootValue());
+        static::assertSame($this->getTreeRootValueOfRootNode($category3), $category3->getTreeRootValue());
+        static::assertSame($this->getTreeRootValueOfRootNode($category4), $category4->getTreeRootValue());
+
+        // Remove
+        $this->em->remove($category);
+        $this->em->remove($category2);
+        $this->em->flush();
+
+        $result = $this->em->createQueryBuilder()->select('c')->from(MPCategoryUuid::class, 'c')->getQuery()->getResult();
+
+        $firstResult = $result[0];
+
+        static::assertCount(1, $result);
+        static::assertSame('4', $firstResult->getTitle());
+        static::assertSame(1, $firstResult->getLevel());
+        static::assertSame($this->getTreeRootValueOfRootNode($firstResult), $firstResult->getTreeRootValue());
+    }
+
+    protected function getUsedEntityFixtures(): array
+    {
+        return [
+            MPCategoryUuid::class,
+        ];
+    }
+
+    private function createCategory(): MPCategoryUuid
+    {
+        $class = MPCategoryUuid::class;
+
+        return new $class();
+    }
+
+    /**
+     * @param array<int|string, int|string|UuidV4|null> $sources
+     */
+    private function generatePath(array $sources): string
+    {
+        $path = '';
+
+        foreach ($sources as $id) {
+            $path .= $id.$this->config['path_separator'];
```

Comment:
This depends on `path_ends_with_separator` in the listener configuration, there's also a `path_starts_with_separator` option and both are used in [AbstractMaterializedPath](https://github.com/doctrine-extensions/DoctrineExtensions/blob/09c61fd9149cb1eb3bfb61edb2604e01df8624d2/src/Tree/Strategy/AbstractMaterializedPath.php#L202). I don't think it can be bypassed without side effects.
I updated `generatePath() `to use the same settings retrieved from the listener configuration.